src/blocks/weather/met_no.rs

```diff
@@ -283,11 +301,48 @@ impl WeatherProvider for Service<'_> {
             })
         };
 
+        let (sunset, sunrise) = match need_sunrise_and_sunset {
+            true => {
+                let sun_query_string: HashMap<&str, String> = map! {
+                    "lat" => &lat,
+                    "lon" => &lon,
+                };
+
+                let sun_data: SunResponse = REQWEST_CLIENT
+                    .get(SUN_URL)
+                    .query(&sun_query_string)
+                    .header(reqwest::header::CONTENT_TYPE, "application/json")
+                    .send()
+                    .await
+                    .error("Forecast request failed")?
+                    .json()
+                    .await
+                    .error("Forecast request failed")?;
```

Comment:
I wonder how far off  the results are to what we can compute ourselves (https://docs.rs/sunrise/)

Reply:
https://api.met.no/weatherapi/sunrise/3.0/sun?lat=40.43&lon=-74.01&date=2024-09-07

```json
{
  "copyright": "MET Norway",
  "licenseURL": "https://api.met.no/license_data.html",
  "type": "Feature",
  "geometry": {
    "type": "Point",
    "coordinates": [-74, 40.4]
  },
  "when": {
    "interval": [
      "2024-09-07T04:53:00Z",
      "2024-09-08T04:56:00Z"
    ]
  },
  "properties": {
    "body": "Sun",
    "sunrise": {
      "time": "2024-09-07T10:29+00:00",
      "azimuth": 81.67
    },
    "sunset": {
      "time": "2024-09-07T23:17+00:00",
      "azimuth": 278.07
    },
    "solarnoon": {
      "time": "2024-09-07T16:53+00:00",
      "disc_centre_elevation": 55.29,
      "visible": true
    },
    "solarmidnight": {
      "time": "2024-09-07T04:53+00:00",
      "disc_centre_elevation": -43.72,
      "visible": false
    }
  }
}
```

```rust
use chrono::{Datelike, TimeZone, Utc};
use chrono_tz::Tz;
use sunrise::sunrise_sunset;

fn main() {
    let (lat, long) = (40.43, -74.01);
    let tz: Tz = tz_search::lookup(lat, long).unwrap().parse().unwrap();
    let date = Utc::now().with_timezone(&tz);
    let (sunrise, sunset) = sunrise_sunset(lat, long, date.year(), date.month(), date.day());
    println!(
        "{}\n{}\n{}\n{}",
        date.format("%Y-%m-%d"),
        tz,
        Utc.timestamp_opt(sunrise, 0).unwrap(),
        Utc.timestamp_opt(sunset, 0).unwrap()
    );
}
```

```
2024-09-07
America/New_York
2024-09-07 10:29:42 UTC
2024-09-07 23:18:23 UTC
```

The results are quite close (for the one random lat/long that I tested)!

Reply:
So we can save an API call _and_ it is simpler? Nice